### PR TITLE
Sidebars & Custom Hooks

### PR DESCRIPTION
**Pull request by horsemaker**
- Added Shrinked Sidebar
- Added Expanded Sidebar
- Added Expanded Sidebar Context & Provider
- Added useWindowSize
- Added useLockBodyScroll
- Modified useOnClickOutside

recording

### CODE REVIEW

File: src/frontend/hooks/useOnClickOutside.js [hunk 1]
```diff
@@ -8,6 +8,11 @@ export const useOnClickOutside = (ref, handler) => {
...
     };
     document.addEventListener("mousedown", listener);
-    return () => document.removeEventListener("mousedown", listener);
+    document.addEventListener("touchstart", listener);
+
+    return () => {
+      document.removeEventListener("mousedown", listener);
+      document.removeEventListener("touchstart", listener);
+    };
   }, [ref, handler]);
```
Comment (MiheerTamkhane) [inline src/frontend/hooks/useOnClickOutside.js:17]: Onclick window hide the sidebar or dropdown. Good work here man,

File: src/frontend/hooks/useWindowSize.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useState, useEffect } from "react";
```
Comment (t007rushi) [inline src/frontend/hooks/useWindowSize.js:1]: Liked this hook implementation

File: src/frontend/layouts/ExpandedSidebar/ExpandedSidebar.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+              className="nav-brand-logo"
+              alt="Kafka"
+            />
+            <h1 className="nav-brand-name">Kafka</h1>
+          </NavLink>
+        </section>
+      </header>
+      <NavLink
+        to="/search"
+        className={({ isActive }) =>
```
Comment (t007rushi) [inline src/frontend/layouts/ExpandedSidebar/ExpandedSidebar.js:78]: Liked the approach

**Review by t007rushi — APPROVED** [review]
Good to merge

**Review by MiheerTamkhane — APPROVED** [review]
Code is good & readable, ui is awsm good to go

**Pull request merged**